src/scripts/reducers/PlayerReducer.js

```diff
@@ -35,7 +38,24 @@ const reducer = {
   [UPDATE_VIDEO_TIME]: (
     state: PlayerRecord,
     action: Action<{ time: number }>
-  ): PlayerRecord => state.set('currentTimeSeconds', action.payload.time),
+  ): PlayerRecord => {
+    const prevPlaybackTimestamp: number = state.get('latestPlaybackTimestamp')
+    const timestamp: number = state.get('isPlaying')
+      ? Date.now()
+      : prevPlaybackTimestamp
+    const prevTotalTimeViewedSeconds: number = state.get(
+      'totalTimeViewedSeconds'
+    )
+
+    return state.merge({
+      currentTimeSeconds: action.payload.time,
+      latestPlaybackTimestamp: timestamp,
+      totalTimeViewedSeconds: prevPlaybackTimestamp
+        ? prevTotalTimeViewedSeconds +
+          (timestamp - prevPlaybackTimestamp) / 1000
+        : 0
+    })
```

Comment:
These changes make it so that we keep track of the total time spent playing the video, regardless of which specific times in the video you have seen. This way we can say "you have been watching this video for 3 seconds"